src/core/agent/services/keriaNotificationService.ts

```diff
@@ -381,34 +395,20 @@ class KeriaNotificationService extends AgentService {
       await this.markNotification(notif.i);
       return false;
     }
-    const rpyRoute = multisigNotification[0].exn.e.rpy.r;
+    const rpyRoute = exchange.exn.e.rpy.r;
     if (rpyRoute === "/end/role/add") {
-      await this.multiSigs.joinAuthorization(multisigNotification[0].exn);
+      await this.multiSigs.joinAuthorization(exchange.exn);
       await this.markNotification(notif.i);
       return false;
     }
     return true;
   }
 
   private async processMultiSigIcpNotification(
-    notif: Notification
+    notif: Notification,
```

Comment:
Quick question on multisig - here we use `groups().getRequest()` instead of `exchanges().get()` - is the response different? I think it is, but we mightn't need the different fields?

Reply:
According to the old logic, we consistently take the first value of the array from` groups.getRequest()`. When logging the returned data, the data contains the necessary fields that are similar to each other (`exn.a`, `exn.e`, ...).

Reply:
Are they exactly the same or just similar?

Reply:
They are just similar, not exactly the same. In the `ExnMessage` type, they are similar in the fields used at layer one of `exn`. The fields within `exn.a` and` exn.e` are different, but since we define `exn.a` and `exn.e` as any, we can currently still use this type interchangeably.

Reply:
If they aren't the same, we cannot just interject one for the other. For multisig messages, can you use `exchanges.get()` like this PR does to check if it's intended for us, and use `groups.getRequest()` within `processMultiSigExnNotification` instead of `exchange`.

Just because the types match with `any` doesn't mean we can interchange...

Reply:
Hmm. Why are we bringing `groups.getRequest()` into `processMultiSigExnNotification`? Originally, the logic for this case only used `exchange.get()`, right?

I just want to confirm: `processMultiSigExnNotification()` will continue using exchange as it is, while I bring `groups.getRequest()` back to `processMultiSigIcpNotification()` and `processMultiSigRpyNotification()`. Does that sound good?

Reply:
Sorry, you're right. `processMultiSigExnNotification` is still exn.

But please revert `processMultiSigIcpNotification` and `processMultiSigRpyNotification` - since they need `groups.getRequest()` they should still call this, and we shouldn't add the `exchange: ExnMessage` parameter to these functions.